instrumentation/restlet/restlet-2.0/library/src/main/java/io/opentelemetry/instrumentation/restlet/v2_0/RestletHeadersGetter.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.restlet.v2_0;
+
+import io.opentelemetry.context.propagation.TextMapGetter;
+import java.util.Locale;
+import org.restlet.Message;
+import org.restlet.Request;
+import org.restlet.util.Series;
+
+final class RestletHeadersGetter implements TextMapGetter<Request> {
+
+  @Override
+  public Iterable<String> keys(Request carrier) {
+    return getHeaders(carrier).getNames();
+  }
+
+  @Override
+  public String get(Request carrier, String key) {
+
+    Series<?> headers = getHeaders(carrier);
+
+    String value = headers.getFirstValue(key);
+    if (value != null) {
+      return value;
+    }
+    return headers.getFirstValue(key.toLowerCase(Locale.ROOT));
```

Comment:
You could use the 2-arg variant instead:
```suggestion
    return headers.getFirstValue(key, /* ignoreCase = */ true);
```